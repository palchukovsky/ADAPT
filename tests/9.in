SCOPE hello {
   DECLARE foo;
   ACCESS foo; // SUCCESS
   DECLARE bar;

   SCOPE hootie {
      DECLARE xyz;
      DECLARE foo; // SUCCESS? -- shadows hello::foo
      ACCESS bar; // SUCCESS
      ACCESS hello::foo; // SUCCESS
   }

   ACCESS hootie::xyz; // SUCCESS
   ACCESS ::hello::bar; // SUCCESS
   DECLARE birdie;
}
ACCESS hello::foo; // SUCCESS
ACCESS ::hello::bar; // SUCCESS