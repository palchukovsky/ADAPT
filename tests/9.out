LINE 3 ACCESS ::hello::foo
LINE 9 ACCESS ::hello::bar
LINE 10 ACCESS ::hello::foo
LINE 13 ACCESS ::hello::hootie::xyz
LINE 14 ACCESS ::hello::bar
LINE 17 ACCESS ::hello::foo
LINE 18 ACCESS ::hello::bar
